### ISSUE TITLE
Impossible to create new folder in upper directory

### ISSUE THREAD

**Issue by konstantinzolotarev** [zed-industries/zed#15823]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

If your project has folder with only one another folder inside like:

```
public
|- images
    |- file.txt
```
![Screenshot 2024-08-05 at 21 14 51](https://github.com/user-attachments/assets/6d5d0e26-1fc5-4c5f-8d9b-7d6a9fc41c41)

It's impossible to create folder under `public` folder. In directory tree it will be `public/images` and new folder will be created only under `public/images`.

![Screenshot 2024-08-05 at 21 19 47](https://github.com/user-attachments/assets/92c1241b-9b64-4561-a968-aee705cb7cc8)

### Environment

Apple Mac M1 Max
DDR: 64GB
MacOS: Sonoma 14.5

**Comment by sleeping-at-night** [comment]
Right click on the folder and select "Unfold Directory". This will make the folders unfold. And if you want folders to fold up again, you can right click and select "Fold Directory".

**Comment by danremollino** [comment]
> Alternatively you can also click on the parent and type `public/test` and it will create `test` in the correctly nested spot.
> 
> If you don't like the folding behavior in the project panel you can disable it in your settings via `project_panel.auto_fold_dirs` [docs](https://zed.dev/docs/configuring-zed#auto-fold-dirs).
> 
> ```
>   "project_panel": {
>     "auto_fold_dirs": false
>   },
> ```

This seems to disable Folding/Unfolding manually. Is this expected?

I also got myself into a state where the option was backwards by manually folding and then disabling `auto_fold_dirs`.

**Comment by notpeter** [comment]
> This seems to disable Folding/Unfolding manually. Is this expected?
>
> I also got myself into a state where the option was backwards by manually folding and then disabling auto_fold_dirs.

There's definitely something funny going on if you change this setting while there's an existing folded directory in the project panel.  Repeated toggling of this setting while expanding/collapsing the folders and I was able observe the same broken/inverted behavior you saw.  I was only able to trigger broken behaviors when I toggle the setting while running Zed and if you reload Zed everything is again fine.

I couldn't come up with a set of steps to reliably reproduce, but would welcome an issue which describes them. (I gave up after a few minutes).  Good find!
